Automattic/sensei
# Add course expiration email

Resolves https://github.com/Automattic/sensei/issues/6471

A part of the ticket issue is implemented here. To test properly we'll also need to checkout the branch https://github.com/Automattic/sensei-pro/pull/2165 of sensei-pro.

## Proposed Changes
* Added a hook for disabling legacy email
* Added function to get the identifier. Will help us when we have multiple emails of a similar type.
* Added patterns for course expiration email
* Moved getting target post id for resume block to be able to reuse it.

## Testing Instructions


1. Enable the Email post type
2. Checkout branch https://github.com/Automattic/sensei-pro/pull/2165 of sensei-pro
3. Make sure you have woocommerce installed (The scheduler runs on it)
4. Re-run the email generator tool
5. Create a course
6. Set the access period setting of that like below
![image](https://user-images.githubusercontent.com/6820724/223725054-4396cefa-57f3-44a7-994a-b8561700d90d.png)
7. Take the course as a student
8. You can tweak the student's course expiration days from here to have more control `http://<local-domain>/wp-admin/admin.php?page=sensei_learners&view=learners&course_id=<Course_ID>#`. You can change the Access Expiration date to any date you want (In our case, 7 days later, 3 days later and Today's date)
![image](https://user-images.githubusercontent.com/6820724/223725716-cc5a3852-fdd6-46e4-88b4-10699c011e95.png)
9. Now go to WooCommerce -> Status -> Scheduled Actions
10. Find the schedules named `sensei_wc_paid_courses_course_expiration_notification_recurring` (search sensei in the searchbox.
11. You'll find 3 events with this name, one with 7 days, one with 3, another with 0. Click 'Run' on the one with 7 days (if your course expires in 7 days)
![image](https://user-images.githubusercontent.com/6820724/223727016-46eb21e0-bd74-46bd-b5f0-ab8847a658c2.png)
12. The page will refresh after clicking run, and will spawn another hook. Now just refresh the page again without clicking anywhere.
![image](https://user-images.githubusercontent.com/6820724/223727597-f458f938-76ae-4f4e-9073-294fcc5f0001.png)
13. Check if the mail is sent.
14. Repeat for 3 days and today's date

## New/Updated Hooks


* `sensei_disable_legacy_emails` - Allows hooking more actions upon disabling the legacy emails.


## Pre-Merge Checklist
